packages/hash/api/src/seed-data/seed-pages.ts

```diff
@@ -0,0 +1,73 @@
+import { Logger } from "@hashintel/hash-backend-utils/logger";
+import { GraphApi } from "@hashintel/hash-graph-client";
+import { PageModel } from "../model";
+
+const createNestedPages = async (
+  owningActorId: string,
+  [top, nested]: [string, string],
+  prevIndex: string | undefined,
+  sharedParams: {
+    graphApi: GraphApi;
+    logger: Logger;
+  },
+): Promise<PageModel> => {
+  const { graphApi } = sharedParams;
+
+  const topPageModel = await PageModel.createPage(graphApi, {
+    actorId: owningActorId,
+    ownedById: owningActorId,
+    title: top,
+    prevIndex,
+  });
+
+  const nestedPageModel = await PageModel.createPage(graphApi, {
+    actorId: owningActorId,
+    ownedById: owningActorId,
+    title: nested,
+  });
+
+  await nestedPageModel.setParentPage(graphApi, {
+    actorId: owningActorId,
+    parentPageModel: topPageModel,
+    prevIndex: topPageModel.getIndex() ?? null,
+    nextIndex: null,
+  });
+
+  return topPageModel;
+};
+
+export type PageList = (string | [string, string])[];
```

Comment:
**Blocking:** I'd prefer a data structure which makes it more obvious what the values are, to make it more maintainable and easier to understand without looking at how the type is used in the code. For instance, something like this:
```TypeScript
type PageDefinition = {
  title: string;
  nestedPages: PageDefinition[];
}
```

This would mean the `seedPages` method would need to be written in a more generic way. This would have the benefit that if we wanted to seed pages with a page hierarchy of 3 levels (as we did previously in `main`), the `seedPages` method wouldn't have to be re-implemented and `PageList` definition reworked.

Reply:
Thanks, addressed in c54bae8fad2e4fa61d5bbe9779ea3fdda975b145
Did simplify things a fair bit.